extensions/ql-vscode/src/run-remote-query.ts

```diff
@@ -59,19 +96,32 @@ export async function runRemoteQuery(cliServer: cli.CodeQLCliServer, credentials
   const query = await fs.readFile(queryFile, 'utf8');
 
   const repositoriesFile = queryFile.substring(0, queryFile.length - '.ql'.length) + '.repositories';
-  if (!(await fs.pathExists(repositoriesFile))) {
-    void showAndLogErrorMessage(`Missing file: '${repositoriesFile}' to specify the repositories to run against. This file must be a sibling of ${queryFile}.`);
-    return;
-  }
+  let ref: string | undefined;
+  let language: string | undefined;
+  let repositories: string[] | undefined;
+
+  // If the user has an explicit `.repositories` file, use that.
+  // Otherwise, prompt user to select repositories from the `codeQL.remoteRepositoryLists` setting.
+  if (await fs.pathExists(repositoriesFile)) {
+    void logger.log(`Found '${repositoriesFile}'. Using information from that file to run ${queryFile}.`);
 
-  const config = yaml.safeLoad(await fs.readFile(repositoriesFile, 'utf8')) as Config;
+    const config = yaml.safeLoad(await fs.readFile(repositoriesFile, 'utf8')) as Config;
 
-  const ref = config.ref || 'main';
-  const language = config.language || await findLanguage(cliServer, uri);
-  const repositories = config.repositories;
+    ref = config.ref || 'main';
```

Comment:
Not something to handle now, but this means that users can only choose a ref if they have a .repositories file. If there is no file, then they can only choose `main`. This is fine for now, but we should consider making this more flexible this later.

Reply:
👍🏽 I expect that we won't need the `ref` at all eventually, though I will double check with the others 😃 
It just specifies from which branch of the controller repository we're running the "remote query" workflow. I can't see why it should ever be different from `main` (except for testing).

Reply:
(Confirmed with @robertbrignull that we don't need to worry about easily setting the `ref` in VS Code. If required for testing, we can send a manual API request.)